pkg/services/query/query.go

```diff
@@ -149,6 +151,19 @@ func (s *Service) handleQueryData(ctx context.Context, user *models.SignedInUser
 		req.Headers[k] = v
 	}
 
+	if parsedReq.httpRequest != nil && parsedReq.httpRequest.Header.Get("Cookie") != "" && ds.JsonData != nil {
```

Comment:
I wonder is now a good opportunity to refactor so that resource, query and proxy calls both use the same code?

Reply:
Good question. They are different though, not sure how easy that would be. Do you have an idea?